<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>930</width>
    <height>620</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>QGifer</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QGridLayout" name="gridLayout">
    <item row="2" column="2">
     <widget class="QMultiSlider" name="multiSlider" native="true">
      <property name="minimumSize">
       <size>
        <width>0</width>
        <height>16</height>
       </size>
      </property>
      <property name="maximumSize">
       <size>
        <width>16777215</width>
        <height>16</height>
       </size>
      </property>
     </widget>
    </item>
    <item row="0" column="0" rowspan="3">
     <widget class="QToolBox" name="toolBox">
      <property name="sizePolicy">
       <sizepolicy hsizetype="Fixed" vsizetype="Preferred">
        <horstretch>0</horstretch>
        <verstretch>0</verstretch>
       </sizepolicy>
      </property>
      <property name="minimumSize">
       <size>
        <width>270</width>
        <height>0</height>
       </size>
      </property>
      <property name="maximumSize">
       <size>
        <width>270</width>
        <height>16777215</height>
       </size>
      </property>
      <property name="currentIndex">
       <number>3</number>
      </property>
      <widget class="QWidget" name="page_1">
       <property name="geometry">
        <rect>
         <x>0</x>
         <y>0</y>
         <width>260</width>
         <height>419</height>
        </rect>
       </property>
       <attribute name="label">
        <string>Preview properties</string>
       </attribute>
       <layout class="QGridLayout" name="gridLayout_5">
        <item row="1" column="0" colspan="2">
         <widget class="Line" name="line_4">
          <property name="orientation">
           <enum>Qt::Horizontal</enum>
          </property>
         </widget>
        </item>
        <item row="3" column="0" colspan="2">
         <widget class="QSlider" name="zoomSlider">
          <property name="minimum">
           <number>10</number>
          </property>
          <property name="maximum">
           <number>100</number>
          </property>
          <property name="value">
           <number>100</number>
          </property>
          <property name="orientation">
           <enum>Qt::Horizontal</enum>
          </property>
         </widget>
        </item>
        <item row="0" column="0">
         <widget class="QLabel" name="label_6">
          <property name="text">
           <string>FPS:</string>
          </property>
         </widget>
        </item>
        <item row="7" column="0">
         <spacer name="verticalSpacer">
          <property name="orientation">
           <enum>Qt::Vertical</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>20</width>
            <height>40</height>
           </size>
          </property>
         </spacer>
        </item>
        <item row="2" column="0">
         <widget class="QLabel" name="zoomLabel">
          <property name="text">
           <string>Zoom:</string>
          </property>
         </widget>
        </item>
        <item row="0" column="1">
         <widget class="QDoubleSpinBox" name="fpsBox"/>
        </item>
        <item row="6" column="0" colspan="2">
         <widget class="QGroupBox" name="groupBox_2">
          <property name="title">
           <string>Loop</string>
          </property>
          <layout class="QGridLayout" name="gridLayout_8">
           <item row="0" column="0">
            <widget class="QRadioButton" name="nlRadio">
             <property name="text">
              <string>Do not loop</string>
             </property>
             <property name="checked">
              <bool>true</bool>
             </property>
            </widget>
           </item>
           <item row="1" column="0">
            <widget class="QRadioButton" name="laRadio">
             <property name="text">
              <string>Loop all</string>
             </property>
            </widget>
           </item>
           <item row="2" column="0">
            <widget class="QRadioButton" name="ssRadio">
             <property name="text">
              <string>Loop from start to stop frame</string>
             </property>
            </widget>
           </item>
          </layout>
         </widget>
        </item>
        <item row="5" column="0" colspan="2">
         <widget class="QCheckBox" name="ratioBox">
          <property name="text">
           <string>Keep aspect ratio</string>
          </property>
         </widget>
        </item>
        <item row="4" column="0" colspan="2">
         <widget class="QCheckBox" name="smoothBox">
          <property name="text">
           <string>Smooth preview</string>
          </property>
         </widget>
        </item>
       </layout>
      </widget>
      <widget class="QWidget" name="page">
       <property name="geometry">
        <rect>
         <x>0</x>
         <y>0</y>
         <width>260</width>
         <height>419</height>
        </rect>
       </property>
       <attribute name="label">
        <string>Margins</string>
       </attribute>
       <layout class="QGridLayout" name="gridLayout_7">
        <item row="0" column="0">
         <widget class="QCheckBox" name="marginBox">
          <property name="text">
           <string>Use margins</string>
          </property>
         </widget>
        </item>
        <item row="1" column="0">
         <widget class="QGroupBox" name="groupBox">
          <property name="title">
           <string>Margins</string>
          </property>
          <layout class="QGridLayout" name="gridLayout_6">
           <item row="1" column="0">
            <widget class="QLabel" name="label_8">
             <property name="text">
              <string>Right</string>
             </property>
            </widget>
           </item>
           <item row="2" column="1">
            <widget class="QSpinBox" name="leftSpin">
             <property name="suffix">
              <string> px</string>
             </property>
             <property name="maximum">
              <number>9999</number>
             </property>
             <property name="value">
              <number>10</number>
             </property>
            </widget>
           </item>
           <item row="1" column="1">
            <widget class="QSpinBox" name="rightSpin">
             <property name="suffix">
              <string> px</string>
             </property>
             <property name="maximum">
              <number>9999</number>
             </property>
             <property name="value">
              <number>10</number>
             </property>
            </widget>
           </item>
           <item row="2" column="0">
            <widget class="QLabel" name="label_9">
             <property name="text">
              <string>Left</string>
             </property>
            </widget>
           </item>
           <item row="3" column="0">
            <widget class="QLabel" name="label_10">
             <property name="text">
              <string>Bottom</string>
             </property>
            </widget>
           </item>
           <item row="3" column="1">
            <widget class="QSpinBox" name="bottomSpin">
             <property name="suffix">
              <string> px</string>
             </property>
             <property name="maximum">
              <number>9999</number>
             </property>
             <property name="value">
              <number>10</number>
             </property>
            </widget>
           </item>
           <item row="0" column="0">
            <widget class="QLabel" name="label_7">
             <property name="text">
              <string>Top:</string>
             </property>
            </widget>
           </item>
           <item row="0" column="1">
            <widget class="QSpinBox" name="topSpin">
             <property name="suffix">
              <string> px</string>
             </property>
             <property name="maximum">
              <number>9999</number>
             </property>
             <property name="value">
              <number>10</number>
             </property>
            </widget>
           </item>
           <item row="4" column="0">
            <spacer name="verticalSpacer_2">
             <property name="orientation">
              <enum>Qt::Vertical</enum>
             </property>
             <property name="sizeHint" stdset="0">
              <size>
               <width>20</width>
               <height>40</height>
              </size>
             </property>
            </spacer>
           </item>
          </layout>
         </widget>
        </item>
       </layout>
      </widget>
      <widget class="QWidget" name="page_3">
       <property name="geometry">
        <rect>
         <x>0</x>
         <y>0</y>
         <width>260</width>
         <height>419</height>
        </rect>
       </property>
       <attribute name="label">
        <string>Balance</string>
       </attribute>
       <layout class="QGridLayout" name="gridLayout_2">
        <item row="1" column="0">
         <widget class="QLabel" name="redLabel">
          <property name="text">
           <string>Red:</string>
          </property>
         </widget>
        </item>
        <item row="2" column="0">
         <widget class="QSlider" name="redSlider">
          <property name="minimum">
           <number>-255</number>
          </property>
          <property name="maximum">
           <number>255</number>
          </property>
          <property name="orientation">
           <enum>Qt::Horizontal</enum>
          </property>
         </widget>
        </item>
        <item row="4" column="0">
         <widget class="QSlider" name="greenSlider">
          <property name="minimum">
           <number>-255</number>
          </property>
          <property name="maximum">
           <number>255</number>
          </property>
          <property name="orientation">
           <enum>Qt::Horizontal</enum>
          </property>
         </widget>
        </item>
        <item row="3" column="0">
         <widget class="QLabel" name="greenLabel">
          <property name="text">
           <string>Green:</string>
          </property>
         </widget>
        </item>
        <item row="6" column="0">
         <widget class="QSlider" name="blueSlider">
          <property name="minimum">
           <number>-255</number>
          </property>
          <property name="maximum">
           <number>255</number>
          </property>
          <property name="orientation">
           <enum>Qt::Horizontal</enum>
          </property>
         </widget>
        </item>
        <item row="5" column="0">
         <widget class="QLabel" name="blueLabel">
          <property name="text">
           <string>Blue:</string>
          </property>
         </widget>
        </item>
        <item row="8" column="0">
         <spacer name="verticalSpacer_3">
          <property name="orientation">
           <enum>Qt::Vertical</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>20</width>
            <height>40</height>
           </size>
          </property>
         </spacer>
        </item>
        <item row="0" column="0">
         <widget class="QCheckBox" name="balanceBox">
          <property name="text">
           <string>Show on preview</string>
          </property>
         </widget>
        </item>
        <item row="7" column="0">
         <widget class="QPushButton" name="resetBalanceButton">
          <property name="text">
           <string>Reset</string>
          </property>
         </widget>
        </item>
       </layout>
      </widget>
      <widget class="QWidget" name="page_2">
       <property name="geometry">
        <rect>
         <x>0</x>
         <y>0</y>
         <width>260</width>
         <height>419</height>
        </rect>
       </property>
       <attribute name="label">
        <string>Output properties</string>
       </attribute>
       <layout class="QGridLayout" name="gridLayout_3">
        <item row="6" column="0" colspan="3">
         <widget class="Line" name="line_2">
          <property name="orientation">
           <enum>Qt::Horizontal</enum>
          </property>
         </widget>
        </item>
        <item row="1" column="0">
         <widget class="QLabel" name="label_2">
          <property name="text">
           <string>Stop frame:</string>
          </property>
         </widget>
        </item>
        <item row="4" column="0">
         <widget class="QLabel" name="label_4">
          <property name="text">
           <string>Height:</string>
          </property>
         </widget>
        </item>
        <item row="11" column="0" colspan="3">
         <widget class="QScrollArea" name="scrollArea">
          <property name="minimumSize">
           <size>
            <width>0</width>
            <height>160</height>
           </size>
          </property>
          <property name="horizontalScrollBarPolicy">
           <enum>Qt::ScrollBarAlwaysOff</enum>
          </property>
          <property name="widgetResizable">
           <bool>true</bool>
          </property>
          <property name="alignment">
           <set>Qt::AlignCenter</set>
          </property>
          <widget class="QWidget" name="scrollAreaWidgetContents">
           <property name="geometry">
            <rect>
             <x>0</x>
             <y>0</y>
             <width>246</width>
             <height>182</height>
            </rect>
           </property>
           <layout class="QGridLayout" name="gridLayout_4">
            <item row="0" column="0">
             <widget class="PaletteWidget" name="paletteWidget" native="true"/>
            </item>
           </layout>
          </widget>
         </widget>
        </item>
        <item row="10" column="0" colspan="3">
         <widget class="QCheckBox" name="autoPaletteBox">
          <property name="text">
           <string>Auto-update form start frame</string>
          </property>
         </widget>
        </item>
        <item row="8" column="0">
         <widget class="QLabel" name="label_5">
          <property name="text">
           <string>Palette size:</string>
          </property>
         </widget>
        </item>
        <item row="0" column="0">
         <widget class="QLabel" name="label">
          <property name="text">
           <string>Start frame:</string>
          </property>
         </widget>
        </item>
        <item row="9" column="0" colspan="3">
         <widget class="QPushButton" name="updatePalButton">
          <property name="text">
           <string>Update from current frame</string>
          </property>
         </widget>
        </item>
        <item row="1" column="1">
         <widget class="QSpinBox" name="stopBox">
          <property name="maximum">
           <number>0</number>
          </property>
         </widget>
        </item>
        <item row="3" column="0">
         <widget class="QLabel" name="label_3">
          <property name="text">
           <string>Width:</string>
          </property>
         </widget>
        </item>
        <item row="2" column="0" colspan="3">
         <widget class="Line" name="line">
          <property name="orientation">
           <enum>Qt::Horizontal</enum>
          </property>
         </widget>
        </item>
        <item row="3" column="1">
         <widget class="QSpinBox" name="widthBox">
          <property name="suffix">
           <string> px</string>
          </property>
          <property name="minimum">
           <number>1</number>
          </property>
          <property name="maximum">
           <number>9999</number>
          </property>
          <property name="value">
           <number>320</number>
          </property>
         </widget>
        </item>
        <item row="4" column="1">
         <widget class="QSpinBox" name="heightBox">
          <property name="suffix">
           <string> px</string>
          </property>
          <property name="minimum">
           <number>1</number>
          </property>
          <property name="maximum">
           <number>9999</number>
          </property>
          <property name="value">
           <number>240</number>
          </property>
         </widget>
        </item>
        <item row="3" column="2" rowspan="2">
         <widget class="QCheckBox" name="whRatioBox">
          <property name="text">
           <string>Keep ratio</string>
          </property>
         </widget>
        </item>
        <item row="8" column="1">
         <widget class="QSpinBox" name="paletteBox">
          <property name="suffix">
           <string> bit</string>
          </property>
          <property name="minimum">
           <number>2</number>
          </property>
          <property name="maximum">
           <number>8</number>
          </property>
          <property name="value">
           <number>8</number>
          </property>
         </widget>
        </item>
        <item row="0" column="1">
         <widget class="QSpinBox" name="startBox">
          <property name="maximum">
           <number>0</number>
          </property>
         </widget>
        </item>
        <item row="5" column="0" colspan="3">
         <widget class="QPushButton" name="getWHButton">
          <property name="text">
           <string>From current frame</string>
          </property>
         </widget>
        </item>
       </layout>
      </widget>
     </widget>
    </item>
    <item row="0" column="2" rowspan="2">
     <widget class="FramePlayer" name="player" native="true">
      <property name="sizePolicy">
       <sizepolicy hsizetype="Expanding" vsizetype="Preferred">
        <horstretch>0</horstretch>
        <verstretch>0</verstretch>
       </sizepolicy>
      </property>
     </widget>
    </item>
    <item row="0" column="1" rowspan="3">
     <widget class="Line" name="line_3">
      <property name="orientation">
       <enum>Qt::Vertical</enum>
      </property>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>930</width>
     <height>20</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuPlayer">
    <property name="title">
     <string>&amp;Player</string>
    </property>
    <addaction name="actionPlay"/>
    <addaction name="actionPause"/>
    <addaction name="actionStop"/>
    <addaction name="separator"/>
    <addaction name="actionNextFrame"/>
    <addaction name="actionPrevFrame"/>
   </widget>
   <widget class="QMenu" name="menu_Help">
    <property name="title">
     <string>&amp;About</string>
    </property>
    <addaction name="actionAbout"/>
   </widget>
   <widget class="QMenu" name="menuEdit">
    <property name="title">
     <string>&amp;Edit</string>
    </property>
    <addaction name="actionExtractGif"/>
    <addaction name="separator"/>
    <addaction name="actionUpdatePalette"/>
    <addaction name="separator"/>
    <addaction name="actionSetAsStart"/>
    <addaction name="actionSetAsStop"/>
   </widget>
   <widget class="QMenu" name="menu_Program">
    <property name="title">
     <string>&amp;Program</string>
    </property>
    <addaction name="actionOpenVideo"/>
    <addaction name="actionOptimizer"/>
    <addaction name="actionConverter"/>
    <addaction name="separator"/>
    <addaction name="actionExit"/>
   </widget>
   <addaction name="menu_Program"/>
   <addaction name="menuEdit"/>
   <addaction name="menuPlayer"/>
   <addaction name="menu_Help"/>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
  <widget class="QToolBar" name="mainToolBar">
   <property name="windowTitle">
    <string>toolBar_2</string>
   </property>
   <property name="iconSize">
    <size>
     <width>32</width>
     <height>32</height>
    </size>
   </property>
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
   <addaction name="actionOpenVideo"/>
   <addaction name="actionExtractGif"/>
   <addaction name="separator"/>
   <addaction name="actionUpdatePalette"/>
   <addaction name="actionSetAsStart"/>
   <addaction name="actionSetAsStop"/>
  </widget>
  <widget class="QToolBar" name="playerToolBar">
   <property name="windowTitle">
    <string>toolBar</string>
   </property>
   <property name="iconSize">
    <size>
     <width>32</width>
     <height>32</height>
    </size>
   </property>
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
   <addaction name="actionPlay"/>
   <addaction name="actionPause"/>
   <addaction name="actionPrevFrame"/>
   <addaction name="actionNextFrame"/>
   <addaction name="actionStop"/>
  </widget>
  <action name="actionOpenVideo">
   <property name="icon">
    <iconset resource="../res.qrc">
     <normaloff>:/res/openvideo.png</normaloff>:/res/openvideo.png</iconset>
   </property>
   <property name="text">
    <string>&amp;Open video...</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+Shift+O</string>
   </property>
  </action>
  <action name="actionPlay">
   <property name="icon">
    <iconset resource="../res.qrc">
     <normaloff>:/res/start.png</normaloff>:/res/start.png</iconset>
   </property>
   <property name="text">
    <string>&amp;Play</string>
   </property>
   <property name="shortcut">
    <string>Space</string>
   </property>
  </action>
  <action name="actionPause">
   <property name="icon">
    <iconset resource="../res.qrc">
     <normaloff>:/res/pause.png</normaloff>:/res/pause.png</iconset>
   </property>
   <property name="text">
    <string>Pa&amp;use</string>
   </property>
   <property name="shortcut">
    <string>Esc</string>
   </property>
  </action>
  <action name="actionStop">
   <property name="icon">
    <iconset resource="../res.qrc">
     <normaloff>:/res/stop.png</normaloff>:/res/stop.png</iconset>
   </property>
   <property name="text">
    <string>&amp;Stop</string>
   </property>
   <property name="shortcut">
    <string>Backspace</string>
   </property>
  </action>
  <action name="actionNextFrame">
   <property name="icon">
    <iconset resource="../res.qrc">
     <normaloff>:/res/next.png</normaloff>:/res/next.png</iconset>
   </property>
   <property name="text">
    <string>&amp;Next frame</string>
   </property>
   <property name="shortcut">
    <string>Right</string>
   </property>
  </action>
  <action name="actionPrevFrame">
   <property name="icon">
    <iconset resource="../res.qrc">
     <normaloff>:/res/prev.png</normaloff>:/res/prev.png</iconset>
   </property>
   <property name="text">
    <string>Pre&amp;vious frame</string>
   </property>
   <property name="shortcut">
    <string>Left</string>
   </property>
  </action>
  <action name="actionCloseVideo">
   <property name="icon">
    <iconset resource="../res.qrc">
     <normaloff>:/res/eject.png</normaloff>:/res/eject.png</iconset>
   </property>
   <property name="text">
    <string>&amp;Close video</string>
   </property>
  </action>
  <action name="actionExit">
   <property name="icon">
    <iconset resource="../res.qrc">
     <normaloff>:/res/exit.png</normaloff>:/res/exit.png</iconset>
   </property>
   <property name="text">
    <string>&amp;Exit</string>
   </property>
  </action>
  <action name="actionAbout">
   <property name="icon">
    <iconset resource="../res.qrc">
     <normaloff>:/res/about.png</normaloff>:/res/about.png</iconset>
   </property>
   <property name="text">
    <string>&amp;About QGifer...</string>
   </property>
  </action>
  <action name="actionImportPng">
   <property name="text">
    <string>*.&amp;PNG files...</string>
   </property>
  </action>
  <action name="actionYoutube">
   <property name="text">
    <string>&amp;YouTube video...</string>
   </property>
  </action>
  <action name="actionExtractGif">
   <property name="icon">
    <iconset resource="../res.qrc">
     <normaloff>:/res/extract.png</normaloff>:/res/extract.png</iconset>
   </property>
   <property name="text">
    <string>E&amp;xtract GIF...</string>
   </property>
  </action>
  <action name="actionUpdatePalette">
   <property name="icon">
    <iconset resource="../res.qrc">
     <normaloff>:/res/palette.png</normaloff>:/res/palette.png</iconset>
   </property>
   <property name="text">
    <string>Update &amp;palette</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+P</string>
   </property>
  </action>
  <action name="actionSetAsStart">
   <property name="icon">
    <iconset resource="../res.qrc">
     <normaloff>:/res/setasstart.png</normaloff>:/res/setasstart.png</iconset>
   </property>
   <property name="text">
    <string>&amp;Start from current frame</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+B</string>
   </property>
  </action>
  <action name="actionSetAsStop">
   <property name="icon">
    <iconset resource="../res.qrc">
     <normaloff>:/res/setasstop.png</normaloff>:/res/setasstop.png</iconset>
   </property>
   <property name="text">
    <string>S&amp;top at current frame</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+E</string>
   </property>
  </action>
  <action name="action_Export_palette">
   <property name="text">
    <string>&amp;Export palette...</string>
   </property>
  </action>
  <action name="actionI_mport_palette">
   <property name="text">
    <string>I&amp;mport palette...</string>
   </property>
  </action>
  <action name="actionOptimizer">
   <property name="icon">
    <iconset resource="../res.qrc">
     <normaloff>:/res/optimizer.png</normaloff>:/res/optimizer.png</iconset>
   </property>
   <property name="text">
    <string>GIF &amp;optimizer...</string>
   </property>
  </action>
  <action name="actionConverter">
   <property name="icon">
    <iconset resource="../res.qrc">
     <normaloff>:/res/converter.png</normaloff>:/res/converter.png</iconset>
   </property>
   <property name="text">
    <string>MJPEG converter...</string>
   </property>
  </action>
 </widget>
 <customwidgets>
  <customwidget>
   <class>FramePlayer</class>
   <extends>QWidget</extends>
   <header>frameplayer.h</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>PaletteWidget</class>
   <extends>QWidget</extends>
   <header>palettewidget.h</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>QMultiSlider</class>
   <extends>QWidget</extends>
   <header>qmultislider.h</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <resources>
  <include location="../res.qrc"/>
 </resources>
 <connections/>
</ui>
